src/EventHandler.cpp

```diff
@@ -516,7 +516,11 @@ void EventHandler::releaseMouse()
  */
 void EventHandler::warpMouse(int x, int y)
 {
-	if (underlyingWindow) { SDL_WarpMouseInWindow(underlyingWindow, x, y); }
+	if (underlyingWindow)
+	{
+		mMouseMotionEvent.emit(x, y, 0, 0);
+		SDL_WarpMouseInWindow(underlyingWindow, x, y);
```

Comment:
I'm wondering if the order of these two lines should be swapped. It probably doesn't matter, though conceptually I would have raised the event after the action took place. That's what I would have expected anyway.